### PR TITLE
Provide better error message when ephemeral download response is not successful

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

File: core/src/ephemeral_server/mod.rs [hunk 1]
```diff
@@ -463,15 +463,19 @@ impl EphemeralExe {
...
                     .send()
-                    .await?
-                    .json()
                     .await?;
+                let info: DownloadInfo = match resp.error_for_status() {
+                    Ok(r) => r.json().await?,
+                    Err(e) => {
+                        return Err(anyhow::Error::from(e));
+                    }
+                };
```
Comment (josh-berry) [inline core/src/ephemeral_server/mod.rs:478]: For patterns like:

```rs
match some_result {
  Ok(_) => ...,
  Err(e) => return Err(...),
}
```

I suggest using [.map_err(...)?](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err) instead; I find it a lot cleaner to read.

**Review by Quinn-With-Two-Ns — APPROVED** [review]

File: core/src/ephemeral_server/mod.rs [hunk 2]
```diff
@@ -591,10 +595,16 @@ async fn download_and_extract(
...
+    // Only stream on successful responses.
+    let stream = match resp.error_for_status() {
+        // We have to map the error type to an io error
+        Ok(resp) => resp
+            .bytes_stream()
+            .map(|item| item.map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, err))),
+        Err(e) => {
+            return Err(anyhow::Error::from(e));
+        }
+    };
```
Comment (josh-berry) [inline core/src/ephemeral_server/mod.rs:607]: Same as above

**Review by josh-berry — APPROVED** [review]
Non-blocking style comments only

File: core/src/ephemeral_server/mod.rs (hunk 1 above)
Comment (bergundy) [reply core/src/ephemeral_server/mod.rs:478]: Yeah, I also prefer that but the result type is different.

Comment (bergundy) [reply core/src/ephemeral_server/mod.rs:478]: I'd need to cast from `Result<Response>` to `anyhow::Result<PathBuf>`.
Same applies below but there's a different return type.

Comment (bergundy) [reply core/src/ephemeral_server/mod.rs:478]: Hmm.. I think I figured out a better way to write it.

**Review by josh-berry — APPROVED** [review]
Oh nice, yeah, that's even better!

**Pull request merged**